diff --git a/node_modules/@tarojs/runtime/dist/bom/raf.js b/node_modules/@tarojs/runtime/dist/bom/raf.js
index 43c6c3e..fbaa96a 100644
--- a/node_modules/@tarojs/runtime/dist/bom/raf.js
+++ b/node_modules/@tarojs/runtime/dist/bom/raf.js
@@ -17,12 +17,12 @@ let now;
 let lastTime = 0;
 // https://gist.github.com/paulirish/1579671
 // https://gist.github.com/jalbam/5fe05443270fa6d8136238ec72accbc0
-const _raf = typeof requestAnimationFrame !== 'undefined' && requestAnimationFrame !== null ? requestAnimationFrame : function (callback) {
+const _raf = process.env.TARO_PLATFORM === 'web' ? requestAnimationFrame : function (callback) {
     const _now = now();
     const nextTime = Math.max(lastTime + 16, _now); // First time will execute it immediately but barely noticeable and performance is gained.
     return setTimeout(function () { callback(lastTime = nextTime); }, nextTime - _now);
 };
-const _caf = typeof cancelAnimationFrame !== 'undefined' && cancelAnimationFrame !== null
+const _caf = process.env.TARO_PLATFORM === 'web'
     ? cancelAnimationFrame
     : function (seed) {
         // fix https://github.com/NervJS/taro/issues/7749
diff --git a/node_modules/@tarojs/runtime/dist/runtime.esm.js b/node_modules/@tarojs/runtime/dist/runtime.esm.js
index 7905f77..60b5a4e 100644
--- a/node_modules/@tarojs/runtime/dist/runtime.esm.js
+++ b/node_modules/@tarojs/runtime/dist/runtime.esm.js
@@ -3784,12 +3784,12 @@ let now;
 let lastTime = 0;
 // https://gist.github.com/paulirish/1579671
 // https://gist.github.com/jalbam/5fe05443270fa6d8136238ec72accbc0
-const _raf = typeof requestAnimationFrame !== 'undefined' && requestAnimationFrame !== null ? requestAnimationFrame : function (callback) {
+const _raf = process.env.TARO_PLATFORM === 'web' ? requestAnimationFrame : function (callback) {
     const _now = now();
     const nextTime = Math.max(lastTime + 16, _now); // First time will execute it immediately but barely noticeable and performance is gained.
     return setTimeout(function () { callback(lastTime = nextTime); }, nextTime - _now);
 };
-const _caf = typeof cancelAnimationFrame !== 'undefined' && cancelAnimationFrame !== null
+const _caf = process.env.TARO_PLATFORM === 'web'
     ? cancelAnimationFrame
     : function (seed) {
         // fix https://github.com/NervJS/taro/issues/7749
